src/core/core.js

```diff
@@ -429,12 +436,16 @@ define(function (require) {
     });
 
     // TODO: ???
-    if (document.readyState === 'complete') {
+    
+    if (sync) {
       this._start();
     } else {
-      window.addEventListener('load', this._start.bind(this), false);
+      if (document.readyState === 'complete') {
+        this._start();
```

Comment:
After realizing why tests with code in the ```suite``` block were working I decided against iframes.  Instead I updated the existing tests to be more robust (they check if ```graphicsContext``` is defined as opposed to try to call methods that use it) and added tests that check the async cases.